src/persistence/messages.ts

```diff
@@ -0,0 +1,107 @@
+import { EnrichedMessage } from "../../generated/definitions/backend/EnrichedMessage";
+import { MessageStatusAttributes } from "../../generated/definitions/backend/MessageStatusAttributes";
+
+type MessageFromBE = EnrichedMessage;
+
+export type MessageOnDB = MessageFromBE & MessageStatusAttributes;
+
+// tslint:disable-next-line: readonly-array no-let
+let inboxMessages: MessageOnDB[] = [];
+// tslint:disable-next-line: readonly-array no-let
+let archivedMessages: MessageOnDB[] = [];
+
+/**
+ * Move the message with ID to the archived collection.
+ * Return false if message ID is not in the Inbox.
+ */
+function archive(id: string): boolean {
+  const index = inboxMessages.findIndex(message => message.id === id);
+  if (index < 0) {
+    return false;
+  }
+  const [toArchived] = inboxMessages.splice(index, 1);
+  const destinationIndex = archivedMessages.findIndex(
+    message => message.id < id
+  );
+  // tslint:disable-next-line: no-object-mutation
+  toArchived.is_archived = true;
+  archivedMessages.splice(destinationIndex, 0, toArchived);
+  return true;
+}
+
+/**
+ * Move the message with ID to the inbox collection.
+ * Return false if message ID is not in the Archive.
+ */
+function unarchive(id: string): boolean {
+  const index = archivedMessages.findIndex(message => message.id === id);
+  if (index < 0) {
+    return false;
+  }
+  const [toInbox] = archivedMessages.splice(index, 1);
+  const destinationIndex = inboxMessages.findIndex(message => message.id < id);
+  // tslint:disable-next-line: no-object-mutation
+  toInbox.is_archived = false;
+  inboxMessages.splice(destinationIndex, 0, toInbox);
+  return true;
+}
+
+/**
+ * Persist a list of messages. The extra attributes is_read and is_archived will
+ * be added.
+ */
+// tslint:disable-next-line: readonly-array no-let
+function persist(messages: MessageFromBE[]): void {
+  inboxMessages = inboxMessages
+    .concat(messages.map(m => ({ ...m, is_read: false, is_archived: false })))
+    .sort((a, b) => (a.id < b.id ? 1 : -1));
+}
+
+function findAllInbox(): ReadonlyArray<MessageOnDB> {
+  return inboxMessages as ReadonlyArray<MessageOnDB>;
```

Comment:
```suggestion
  return inboxMessages;
```
maybe it is no need since it is declared in the return type